src/Tizen.Diagnostics/Tizen.Diagnostics/ExceptionFactory.cs

```diff
@@ -0,0 +1,58 @@
+﻿/*
+* Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+
+namespace Tizen.Diagnostics
+{
+    internal static class ExceptionFactory
+    {
+        internal static Exception CreateException(Interop.Diagnostics.DiagnosticsError err)
+        {
+            Tizen.Log.Error(Interop.Diagnostics.LogTag, $"Error {err}");
+
+            switch (err)
+            {
+                case Interop.Diagnostics.DiagnosticsError.InvalidParameter:
+                    return new ArgumentException("Invalid parameters provided");
+
+                case Interop.Diagnostics.DiagnosticsError.IoError:
+                    return new InvalidOperationException("I/O Error occured");
+
+                case Interop.Diagnostics.DiagnosticsError.OutOfMemory:
+                    return new OutOfMemoryException("Out of memory");
```

Comment:
❌ DO NOT explicitly throw OutOfMemoryException. This exception is to be thrown only by the CLR infrastructure.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/using-standard-exception-types#outofmemoryexception

Reply:
Should I just remove it or replace with other error?
```
case Interop.Diagnostics.DiagnosticsError.OutOfMemory:
    return new InvalidOperationException("I/O Error occured");
```

Reply:
Actually there are quite a few other APIs that are throwing OutOfMemoryException.

Reply:
Yes, should be replaced with InvalidOperationException,

Reply:
Done. However, I'm not sure about the exception message.